samples/host_exerciser/host_exerciser.h

```diff
@@ -0,0 +1,489 @@
+	enum {
+		offset = CSR_AFU_DSM_BASEH
+	};
+	csr_afu_dsm_baseh(uint32_t v) : value(v) {}
+	uint32_t value;
+	struct {
+		uint32_t DsmBaseH : 32;
+	};
+};
+
+union csr_ctl{
+	enum {
+		offset = CSR_CTL
+	};
+
+	csr_ctl(uint32_t v) : value(v) {}
+	uint32_t value;
+	struct {
+		uint32_t Reserved4 : 29;
+		uint8_t ForcedTestCmpl : 1;
+		uint8_t Start : 1;
+		uint8_t ResetL : 1;
+	};
+};
+
+union csr_cfg {
+	enum {
+		offset = CSR_CFG
+	};
+
+	csr_cfg(uint32_t v) : value(v) {}
+	uint32_t value;
+	struct {
+
+		uint32_t RsvdZ1 : 2;
+		uint32_t cr_interrupt_testmode : 1;
+		uint32_t cr_interrupt_on_error : 1;
+		uint32_t cr_test_cfg : 8;
+		uint32_t RsvdZ2 : 1;
+		uint32_t cr_chsel : 2;
+		uint32_t cr_rdsel : 2;
+		uint32_t cr_delay_en : 1;
+
+		uint32_t cr_multiCL_len : 2;
+		uint32_t cr_mode : 3;
+		uint32_t c_cont : 1;
+		uint32_t cr_wrthru_en : 1;
+	};
+};
+
+
+union csr_inact_thresh {
+	enum {
+		offset = CSR_INACT_THRESH
+	};
+
+	csr_inact_thresh(uint32_t v) : value(v) {}
+	uint32_t value;
+	struct {
+
+	uint32_t csr_inact_thresh_value :32;
+	};
+};
+
+union csr_interrupt0 {
+	enum {
+		offset = CSR_INTERRUPT0
+	};
+
+	csr_interrupt0(uint32_t v) : value(v) {}
+	uint32_t value;
+	struct {
+
+	uint32_t VectorNUm : 16;
+	uint32_t apci_id : 16;
+	};
+};
+
+
+union csr_swtest_msg {
+	enum {
+		offset = CSR_SWTEST_MSG
+	};
+	csr_swtest_msg(uint64_t v) : value(v) {}
+	uint64_t value;
+	struct {
+		uint64_t swtest_msg : 64;
+	};
+};
+
+
+union csr_status0 {
+	enum {
+		offset = CSR_STATUS0
+	};
+	csr_status0(uint64_t v) : value(v) {}
+	uint64_t value;
+	struct {
+		uint32_t numReads : 32;
+		uint32_t numWrites : 32;
+	};
+};
+
+
+union csr_status1 {
+	enum {
+		offset = CSR_STATUS1
+	};
+	csr_status1(uint64_t v) : value(v) {}
+	uint64_t value;
+	struct {
+		uint32_t numPendReads : 32;
+		uint32_t numPendWrites : 32;
+	};
+};
+
+
+union csr_error {
+	enum {
+		offset = CSR_ERROR
+	};
+	csr_error(uint64_t v) : value(v) {}
+	uint64_t value;
+	struct {
+		uint32_t RsvdZ1 : 32;
+		uint32_t error : 32;
+	};
+};
+
+
+union csr_error1 {
+	enum {
+		offset = CSR_ERROR1
+	};
+	csr_error1(uint64_t v) : value(v) {}
+	uint64_t value;
+	struct {
+		uint64_t error : 32;
+	};
+};
+
+using test_afu = opae::afu_test::afu;
+using test_command = opae::afu_test::command;
+
+class host_exerciser : public test_afu {
+public:
+	host_exerciser(const char *afu_id = AFU_ID)
+  : test_afu("host_exerciser", afu_id)
+  , count_(1)
+  {
+    app_.add_option("-m,--mode", mode_str_, "mode {lpbk,read, write, trput}")->default_val("lpbk");
+    app_.add_option("-r,--read", read_str_, "read request type{rdline_s, rdline_l,mixed}")->default_val("rdline_s");
+    app_.add_option("-w,--write", write_str_, "write request type {wrline_m, wrline_l}")->default_val("wrline_m");
+    app_.add_option("-d,--delay", delay_, "Enables random delay insertion between requests")->default_val(false);
+    app_.add_option("--multi-cl", multi_cl_, "multi Cache lineone of {0, 1, 3}")->default_val(0);
```

Comment:
similar comment as for line 338 except we wouldn't use a transformer.
Here instead we could just use CLI::IsMember.
```c++
  app_.add_option("--multi-cl", multi_cl_, "multi Cache lineone of {0, 1, 3}")->default_val(0)
    ->check(CLI::IsMember({0, 1, 3}));
```

Also, are the multi cli values are 0, 1, and 3? I thought it was 0, 1, and 2.

Reply:
fixed